### ISSUE TITLE
Add Check for Write access to /tmp + Check for modules php_mbstring, php_zip

### ISSUE THREAD

**Issue by Arnothar** [contao/contao-manager#281]
My environment:

Operation System: Ubuntu 16.04.4 LTS
Webserver: Apache2 (2.4.18)
PHP-Version: 5.6 and 7.2 (both via fpm + unix socket,  see Apache Config Snippet)
Installed PHP7.2 Modules (PHP5.6 modules are installed accordingly:
- php7.2-bcmath
- php7.2-cli
- php7.2-common
- php7.2-curl
- php7.2-fpm
- php7.2-gd
- php7.2-intl
- php7.2-json
- php7.2-mysql
- php7.2-opcache
- php7.2-pgsql
- php7.2-readline
- php7.2-soap
- php7.2-xml

Apache2 Config:
\<VirtualHost *:80\>
    ServerName domain.local
    ServerAlias   www.domain.local
    ServerAdmin admin@webserver
    DocumentRoot /var/www/domain/web
    ErrorLog ${APACHE_LOG_DIR}/domain.local_contao4_error.log
    CustomLog ${APACHE_LOG_DIR}/domain.local__contao4_access.log combined
        \<Directory /\>
                Options FollowSymLinks
                AllowOverride All
        \</Directory\>
        \<FilesMatch \.php$\>
            SetHandler "proxy:unix:/var/run/php/php7.2-fpm.sock|fcgi://localhost/"
        \</FilesMatch\>
\</VirtualHost\>

Chosen Contao Manager Options:
Hoster: Others... + /usr/bin/php7.2 as binary path to php interpreter
Use Composer Cloud: Yes
Contao Version: 4.5

Problem:

The download of Contao fails because of:

1. no write access to /tmp (owner was root:root with 770 and Apache is running under www-data user)
2. missing module mod_mbstring
3. missing module php_zip

The download fails after about 5 minutes  with a simple "Console task terminated" message. No information why it failed. After I tried the same with installing Contao 4.4, I saw the black-green console output and there were PHP interpreter errors about those 3 defects.
Adjusting /tmp to be owned by www-data:root and installing those 2 php modules fixed the problem.

Solution for Contao:
Contao Manager should also check those 3 defects to point out problems with downloads

**Comment by fritzmg** [comment]
A write access check for the system tmp dir is already in the Check: https://github.com/contao/check/blob/master/check/controller/contao4.php#L163-L177

What exactly is "mod_mbstring" for?

php_zip isn't required by Contao directly, but Composer requires it, if no `unzip` shell command is available on the server. The Contao Manager could check for the existence of either the `unzip` command or the availability of `php_zip`.

**Comment by Toflar** [comment]
> The Contao Manager could check for the existence of either the `unzip` command or the availability of `php_zip`.

Correct 👍

**Comment by Toflar** [comment]
> What exactly is "mod_mbstring" for?

It's for multibyte string handling. Any php setup not providing this extension is really strange. It's 2018 and we're talking about a CMS that supports any locale, language etc. There are partial polyfills (https://github.com/symfony/polyfill-mbstring) but still.
Anyway, at the moment it's not required by Contao, nor Composer or the Manager itself, so I wonder why it's needed? Maybe an extension requires it?

**Comment by fritzmg** [comment]
> Anyway, at the moment it's not required by Contao, nor Composer or the Manager itself, so I wonder why it's needed? Maybe an extension requires it?
```
$ composer why ext-mbstring
ausi/slug-generator  v1.0.2  requires  ext-mbstring (*)
beberlei/assert      v2.9.6  requires  ext-mbstring (*)
erusev/parsedown     1.7.1   requires  ext-mbstring (*)
league/uri           5.3.0   requires  ext-mbstring (*)
```

**Comment by Toflar** [comment]
There you go :) As I said, URL handling, slug generation etc. is supported with special characters that take up more than one byte. It's a must for any multilingual CMS. PHP just must be compiled with `intl` and `mbstring` support. I think we should require it in general. /cc @contao/developers

**Comment by aschempp** [comment]
Sounds reasonable. Can we somehow figure out all extensions that are required? But please for Contao 4.4, because that's the minimum we can install.

**Comment by aschempp** [comment]
implemented in version 1.1.0
